src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core'
+import { Segment } from '../../../core/models/segment'
+import { ConfigurationService } from '../../../shared/services/configuration-service'
+import { StudioConfiguration } from '../../../shared/services/studio-configuration'
+import { Tv2VideoClipAction } from '../../../shared/models/tv2-action'
+import { Subscription } from 'rxjs'
+import { ActionService } from '../../../shared/abstractions/action.service'
+import { MediaStateService } from '../../../shared/services/media-state.service'
+import { Media } from '../../../shared/services/media'
+
+@Component({
+  selector: 'sofie-mini-shelf',
+  styleUrls: ['./mini-shelf.component.scss'],
+  templateUrl: './mini-shelf.component.html',
+})
+export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() public segment: Segment
+  @Input() public videoClipAction: Tv2VideoClipAction | undefined
+
+  private readonly defaultAssetForThumbnail: string = 'assets/sofie-logo.svg'
+  protected mediaDuration: number = 0
+  private configurationMediaPreviewUrl: string
+  private configurationServiceSubscription: Subscription
+  private executeActionSubscription: Subscription
+  private serverPostrollDuration: number = 0
+
+  constructor(
+    private readonly actionService: ActionService,
+    private readonly configurationService: ConfigurationService,
+    private readonly mediaStateService: MediaStateService
+  ) {}
+
+  public ngOnInit(): void {
+    this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((configuration: StudioConfiguration) => {
+      this.configurationMediaPreviewUrl = configuration.data.settings.mediaPreviewUrl
+      this.serverPostrollDuration = configuration.data.blueprintConfiguration.ServerPostrollDuration
+    })
+    void this.updateMediaDataDuration()
+  }
+
+  private async updateMediaDataDuration(): Promise<void> {
+    if (this.segment.metadata?.miniShelfVideoClipFile !== undefined) {
+      let media: Media | undefined = await this.mediaStateService.getMedia(this.segment.metadata?.miniShelfVideoClipFile)
+      if (!media || !media.duration) {
+        return
+      }
+      if (media.duration < this.serverPostrollDuration) {
+        this.mediaDuration = this.NaN
+      } else {
+        this.mediaDuration = media.duration - this.serverPostrollDuration
+      }
+    }
+  }
```

Comment:
If you invert the first if-statement you can reduce the nesting to one:
```
private async updateMediaDataDuration(): Promise<void> {
  if (!this.segment.metada?.miniShelfVideoClipFile) {
    return
  }
  ...
}
```

But a lot of the logic in this method seems redundant. If you just store the entire Media object instead of just the duration this would just become:
```
private async fetchMediaForMiniShelf(): Promise<void> {
  if (!...) {  // check if there isn't a miniShelVideoClipFile
    return
  }
  this.media = await this.mediateStateService.getMedia(...)
}
```
And then in your html file your can subtract the ServerPostRollDuration when you are displaying it.

Reply:
I prefer to have the calculus in a method, else html will do calculus and pipe transformations based on value.